### PR TITLE
Add a dependency manager

### PR DESCRIPTION
**Pull request by aymaneMx**
In this PR I :
- Managed the dependencies using `pipenv`
- Reformatted the code to respect the `pep8`

### CODE REVIEW

File: .idea/workspace.xml [hunk 1]
```diff
@@ -2,15 +2,7 @@
 <project version="4">
   <component name="ChangeListManager">
     <list default="true" id="8cf5bb6c-80cc-4311-8b3d-442ab32b9ad4" name="Default Changelist" comment="">
-      <change beforePath="$PROJECT_DIR$/.idea/workspace.xml" beforeDir="false" afterPath="$PROJECT_DIR$/.idea/workspace.xml" afterDir="false" />
```
Comment (MouadBH) [inline .idea/workspace.xml]: Is it necessary to add this file to the project or have you ignored it.

**Review by MouadBH — COMMENTED** [review]
Is it necessary to add this ```.idea/workspace.xml ```  file to the project or have you ignored it.

**Review by aymaneMx — COMMENTED** [review]
nn, it's generated by my IDE (pycharm) .. plz ignore it (I added `.idea/` to `.gitignore` file)

**Comment by MouadBH** [comment]
Thank for your contribution❤ , branch merged.

**Pull request merged**